src/components/common/NumberField/index.tsx

```diff
@@ -10,6 +10,9 @@ export const _formatNumber = (value: string) => {
     return value
   }
 
+  // Replace commas with dots
```

Comment:
I would specualte that the thousand separators are more for visual presentation of a number and not for entering the number in an input field.
![grafik](https://github.com/safe-global/safe-wallet-web/assets/693770/4039a4f8-de47-4d5d-adc2-071ce2134a71)

Google sheet automatically replaces the numpad comma with a dot. The normal comma on the keyboard is left unreplaced, but I think that this has to do with the fact that you can also enter text data in a cell. 
Also if I use commas in a number when I enter it, after I save and try to modify it again the commas are gone:
![grafik](https://github.com/safe-global/safe-wallet-web/assets/693770/c2b19cb2-e890-4ace-80bc-aa85f8c00bb8)

Reply:
btw, 1inch, uniswap etc - they all do the same - they replace comma with a dot.

Reply:
I think it makes sense but then we should not allow adding a dot or a comma more than once. Right now I could type a number like `1.123123.123` which is confusing. It should rather have no thousand separators at all and only one for decimals.

Reply:
I think you are blowing this out of proportion :), but ok. I just wanted to replace the comma - which is often used, but wrong with something that the number input will accept - which is . 

Having multiple dots is clearly an error and is correctly highlighted as error by the component validation. 

Anyway - I now replace all , with . Then I check that we don't have more than one dot. If we do I leave the first one. 

I also remove all non-numbers from the input as it also doesn't make sense to have abcdefg... characters in the string. 

Now with this change you could argue that someone might want to copy paste
100,000,000.23 and will be very surprised to find out that this is 100.00000023 instead of 100000000.23, but hey!

Reply:
To clarify, I agree with the change overall, it’s a good quality of life improvement. I just meant that we should explain in the comment that this is to make both a comma and a dot work as decimal points.

Reply:
I've fixed all conflicts and iterated a bit more

entered amount -> output amount
`` -> ``
`0`-> `0`
`.` -> `0.`
`a123` -> `123`
`123a` -> `123`
`123....123` -> `123`
`123.123.456` -> `123.123456`
`-1` -> `1`
`+1` -> `1`

if we input `,` anywhere in the string it is going to be replaced with `.`. If multiple `,` are entered only one `.` appears.